ios/Handlers/RegistrationHandler.swift

```diff
@@ -76,14 +77,19 @@ extension RegistrationHandler : RegistrationConnectorToHandlerProtocol {
     }
 
     func cancelCustomRegistration() throws {
-        try handleOTPCode(nil, true)
+        guard let customRegistrationChallenge = self.customRegistrationChallenge else {
+            throw WrapperError.actionNotAllowed(description: RegistrationHandler.cancelBrowserRegisrationNotAllowed)
+        }
+        customRegistrationChallenge.sender.cancel(customRegistrationChallenge)
+        handleDidFailToRegister()
     }
 
     func cancelBrowserRegistration() throws {
         guard let browserRegistrationChallenge = self.browserRegistrationChallenge else {
-            throw WrapperError.registrationNotInProgress
+            throw WrapperError.actionNotAllowed(description: RegistrationHandler.cancelCustomRegisrationNotAllowed)
```

Comment:
typo:
`cancelCustomRegisrationNotAllowed`
should be:
`cancelCustomRegistrationNotAllowed`